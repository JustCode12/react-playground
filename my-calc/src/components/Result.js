import React, { useContext } from 'react';
import { CalculationContext } from './CalculationProvider';

const Result = () => {
    const { result } = useContext(CalculationContext);
    return (
        <div><span>Result: </span>{result}</div>
    )
}
export default Result;

