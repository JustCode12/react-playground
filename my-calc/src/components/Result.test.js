import React from 'react';
import { shallow } from 'enzyme';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Result from './Result.js';

configure({ adapter: new Adapter() })

const TestContext = React.createContext();
const TestProvider = () => {
  return (<TestContext.Provider>
    value={{
      result: 5
    }}
  </TestContext.Provider>)
}

describe('Result component test', () => {

  const wrapper = shallow(
    <TestProvider><Result /></TestProvider>
  );
  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});