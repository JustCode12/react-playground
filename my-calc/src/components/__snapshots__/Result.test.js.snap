// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Result component test should match the snapshot 1`] = `ShallowWrapper {}`;
